### DISCUSSION TITLE
Blueretro on Amiga

### DISCUSSION THREAD

**Discussion by ozkanocakli** [darthcloud/BlueRetro#305] [General]
I built blueretro adapter for amiga and ps2,  i have few issues

I used master system schematics for amiga, on dev board it works , on my dedicated pcb i have some problems

1. although esp32 use very low power consumption amiga port does not supply enough power, i have to feed with external power, is there any way using esp32 in low power mode
2. my esp32 adapter sometimes does not power properly on cold power, when i clean with isp alcohol it works again :) is there any pin should i connects to gnd, (same both on amiga and ps2 adapter), what am i doing wrong 
3.My buttons also does not working 1st button pins are connected En To Gnd, 2nd button connected to IO0 To Gnd
4. does ESP32-WROOM-32E, ESP32-WROOM-32U works with blueretro

thanks darthcloud for his efforts

**Comment by darthcloud** [comment]
1. The only way to save power would be to reduce the CPU frequency, while it's not an option for various reason for most platform, I think it wouldn't be an issue for parallel system. I will try to provide a build later.
2. Maybe try to add a PMIC on the EN line as described here:
    https://github.com/darthcloud/BlueRetro/issues/269#issuecomment-1124737460
3.  ESP32-WROOM-32E is what is officially supported, but in theory any should work.
    But module not manufactured by Espressif (clone) are known to have weird issue.

**Reply by darthcloud** [reply]
Here a firmware to test, it's untested btw. Only use the parallel FW, others are likely broken.
[v1.6.1_slow_cpu_test.zip](https://github.com/darthcloud/BlueRetro/files/8912067/v1.6.1_slow_cpu_test.zip)

**Comment by ozkanocakli** [comment]
Hi, adding 100nf between gnd and vcc3.3 to esp32 molue fixes all problems, right now both buttons works and there is no stability problems

I tried your test firmware(reduced cpu frequency) but no luck, power led is getting on but pulse led only lid onces and turn off

**Comment by ozkanocakli** [comment]
And another question what is the minimum flash needs for esp32 version, although firmware size over 512kb does it fit 4 mbit version of esp32

**Reply by darthcloud** [reply]
Any ESP32 size will work. 4MB is the standard size.

**Reply by ozkanocakli** [reply]
After playing around 3 or 4 minutes, controller and amiga loosing sync, does not respond correctly, after loosing sync, i am not touching buttons on ps4 controller and wait than resync again, any suggestions

**Reply by darthcloud** [reply]
I guess you means PS4 disconnect from BlueRetro, since it's parallel there is no sync between BlueRetro and the Amiga.

Make sure you PS4 ctrl is fully charged, if that doesn't help you probably need to power BlueRetro externaly.

**Reply by ozkanocakli** [reply]
i found the sync issue between blueretro adapter and ps4 controller is distance problem, in 1 metre range no problem, getting far from 1 meter sync is dropping and buttons latch on amiga, what is the regular distance between blueretro adapter and bluetooh controller

**Reply by darthcloud** [reply]
I got no problem being 6 meter away, I think you got power issue, use external.

**Comment by ozkanocakli** [comment]
Does blueretro firmware fits on 4 mbit capacity version of esp32
